### PR TITLE
fix(dedicated): hgv2 wording adaptation with the renewal period

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `release/prod-2022-w27` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-9667 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I adapt the wording with the renewal period

## Related
:flags: Translation:
- [ ] TRANS-44761

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.html [hunk 1]
```diff
@@ -16,7 +16,7 @@
 <oui-stepper data-current-index="$ctrl.currentStep">
     <oui-step-form
         data-header="{{:: ('dedicated_server_dashboard_upgrade_select_option_title_' + $ctrl.upgradeType) | translate }}"
-        data-description="{{:: ('dedicated_server_dashboard_upgrade_select_option_description_' + $ctrl.upgradeType) | translate }}"
+        data-description="{{:: $ctrl.renewPeriod !== $ctrl.DEFAULT_INTERVAL ? 'dedicated_server_dashboard_upgrade_select_option_description_' + $ctrl.upgradeType + '_with_interval' : 'dedicated_server_dashboard_upgrade_select_option_description_' + $ctrl.upgradeType | translate : { interval: $ctrl.renewPeriod } }}"
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.html:19]: better into controller i guess

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.routing.js [hunk 2]
```diff
@@ -29,6 +31,15 @@ export default /* @ngInject */ ($stateProvider) => {
...
             getOptionPrice(option1)?.value - getOptionPrice(option2)?.value,
         ),
+      renewPeriod: /* @ngInject */ (upgradeOptions, getOptionInterval) => {
+        let interval = DEFAULT_INTERVAL;
+        upgradeOptions.forEach((option) => {
+          if (getOptionInterval(option) !== DEFAULT_INTERVAL) {
+            interval = getOptionInterval(option);
+          }
+        });
+        return interval;
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.routing.js:41]: const upgradeOption = upgradeOptions.find((option) => getOptionInterval(option) !== DEFAULT_INTERVAL)

return upgradeOption || DEFAULT_INTERVAL;

Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.routing.js:41]: It's not the same. I just want the interval value and not an object with all values.

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.html (hunk 1 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.html]: it's fixed

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.html [hunk 3]
```diff
@@ -39,7 +39,7 @@
                 </oui-radio>
             </oui-datagrid-column>
             <oui-datagrid-column
-                data-title=":: 'dedicated_server_dashboard_upgrade_select_option_price' | translate"
+                data-title=":: $ctrl.renewPeriod !== $ctrl.DEFAULT_INTERVAL ? 'dedicated_server_dashboard_upgrade_select_option_price_with_interval' : 'dedicated_server_dashboard_upgrade_select_option_price' | translate : { interval: $ctrl.renewPeriod }"
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.html]: it's fixed

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -11,9 +11,12 @@
...
   "dedicated_server_dashboard_upgrade_select_option_description_ram": "Les prix indiqués correspondent à un mois complet. Votre première facture pour cette option sera donc calculée au prorata du mois en cours, selon la date de renouvellement de votre serveur.",
   "dedicated_server_dashboard_upgrade_select_option_description_storage": "Les prix indiqués sont pour un mois complet. La première facture que vous recevrez pour cette option sera calculée au prorata de votre mois en cours selon la date de renouvellement de votre serveur.",
+  "dedicated_server_dashboard_upgrade_select_option_description_ram_with_interval": "Les prix indiqués correspondent à une période de {{interval}} mois complets. Votre première facture pour cette option sera donc calculée au prorata de la période en cours, selon la date de renouvellement de votre serveur.",
+  "dedicated_server_dashboard_upgrade_select_option_description_storage_with_interval": "Les prix indiqués sont pour une période de {{interval}} mois complets. La première facture que vous recevrez pour cette option sera calculée au prorata de votre période en cours selon la date de renouvellement de votre serveur.",
```
Comment (JeremyDec) [inline packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/translations/Messages_fr_FR.json:15]: Looks like these sentences are quite the same no? Can it be harmonized?

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.routing.js [hunk 5]
```diff
@@ -29,6 +31,15 @@ export default /* @ngInject */ ($stateProvider) => {
           (option1, option2) =>
             getOptionPrice(option1)?.value - getOptionPrice(option2)?.value,
         ),
+      renewPeriod: /* @ngInject */ (upgradeOptions, getOptionInterval) => {
+        let interval = DEFAULT_INTERVAL;
+        upgradeOptions.forEach((option) => {
+          if (getOptionInterval(option) !== DEFAULT_INTERVAL) {
+            interval = getOptionInterval(option);
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.routing.js:38]: **nitpick:** `getOptionInterval` is called twice. Can we store it in a variable and avoid multiple calls?

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.html [hunk 6]
```diff
@@ -16,7 +16,7 @@
 <oui-stepper data-current-index="$ctrl.currentStep">
     <oui-step-form
         data-header="{{:: ('dedicated_server_dashboard_upgrade_select_option_title_' + $ctrl.upgradeType) | translate }}"
-        data-description="{{:: ('dedicated_server_dashboard_upgrade_select_option_description_' + $ctrl.upgradeType) | translate }}"
+        data-description="{{:: $ctrl.hasRenewPeriod ? 'dedicated_server_dashboard_upgrade_select_option_description_' + $ctrl.upgradeType + '_with_interval' : 'dedicated_server_dashboard_upgrade_select_option_description_' + $ctrl.upgradeType | translate : { interval: $ctrl.renewPeriod } }}"
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.html:19]: regarding `hasRenewPeriod` function, i think you have to use it like:
```suggestion
        data-description="{{:: $ctrl.hasRenewPeriod() ? 'dedicated_server_dashboard_upgrade_select_option_description_' + $ctrl.upgradeType + '_with_interval' : 'dedicated_server_dashboard_upgrade_select_option_description_' + $ctrl.upgradeType | translate : { interval: $ctrl.renewPeriod } }}"
```

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.routing.js (hunk 5 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.routing.js]: It's fixed.

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.routing.js (hunk 2 above)
Comment (mohammed-zahaf) [reply packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.routing.js]: For me we have a no deterministic search.
the loop will iterate and the interval will be set more than one times a think.
So we will take every time the last value that is different from `default`, it's expected ?

Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.routing.js]: As seen together. I applied your method.

**Review by anooparveti — APPROVED** [review]

**Pull request merged**